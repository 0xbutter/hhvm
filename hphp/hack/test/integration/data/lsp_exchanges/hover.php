<?hh  //strict
// comment
function a_hover(): int {
  return b_hover();
}

# A comment describing b_hover.
function b_hover(): int {
  return 42;
}

// A comment describing THE_ANSWER
const int THE_ANSWER = 42;

function return_the_answer(): int {
  return THE_ANSWER;
}

// comment describing the "pad_left" function and its parameters
function pad_left(string $s, int $i, string $pad): string {
  return $s;
}

// comment describing "return_a_string" function
function return_a_string(): string {
  $pad = "hello";
  $x = pad_left("StringToPad", 20, $pad);
  return $x;
}
