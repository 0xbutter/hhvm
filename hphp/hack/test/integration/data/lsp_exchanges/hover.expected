[
  {
    "jsonrpc": "2.0",
    "id": 1,
    "result": {
      "capabilities": {
        "textDocumentSync": {
          "openClose": true,
          "change": 2,
          "willSave": false,
          "willSaveWaitUntil": false,
          "save": {
            "includeText": false
          }
        },
        "hoverProvider": true,
        "completionProvider": {
          "resolveProvider": true,
          "triggerCharacters": [
            "$",
            ">",
            "\\",
            ":",
            "<"
          ]
        },
        "definitionProvider": true,
        "referencesProvider": true,
        "documentHighlightProvider": true,
        "documentSymbolProvider": true,
        "workspaceSymbolProvider": true,
        "codeActionProvider": false,
        "documentFormattingProvider": true,
        "documentRangeFormattingProvider": true,
        "renameProvider": false,
        "typeCoverageProvider": true,
        "rageProvider": true
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 2,
    "result": {
      "contents": [
        {
          "language": "hack",
          "value": "int b_hover"
        }
      ],
      "range": {
        "start": {
          "line": 3,
          "character": 9
        },
        "end": {
          "line": 3,
          "character": 16
        }
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "id": 3,
    "result": null
  },
  {
    "jsonrpc": "2.0",
    "method": "telemetry/event",
    "params": {
      "type": 4,
      "message": "Version in hhconfig=[NoVersion]"
    }
  }
]